table/column.go

```diff
@@ -283,6 +283,12 @@ func handleZeroDatetime(ctx sessionctx.Context, col *model.ColumnInfo, casted ty
 // TODO: change the third arg to TypeField. Not pass ColumnInfo.
 func CastValue(ctx sessionctx.Context, val types.Datum, col *model.ColumnInfo, returnErr, forceIgnoreTruncate bool) (casted types.Datum, err error) {
 	sc := ctx.GetSessionVars().StmtCtx
+	if val.Kind() == types.KindNull && col.FieldType.Tp == mysql.TypeTimestamp {
+		if v, err := expression.GetTimeValue(ctx, strings.ToUpper(ast.CurrentTimestamp), col.Tp, int8(col.Decimal)); err == nil {
```

Comment:
Can you encapsulate the code about getting current timestamp into a function and reuse it directly?
```go
if upperX == strings.ToUpper(ast.CurrentTimestamp) {
    defaultTime, err := getStmtTimestamp(ctx)
    if err != nil {
        return d, err
    }
    value.SetCoreTime(types.FromGoTime(defaultTime.Truncate(time.Duration(math.Pow10(9-int(fsp))) * time.Nanosecond)))
    if tp == mysql.TypeTimestamp || tp == mysql.TypeDatetime {
        err = value.ConvertTimeZone(time.Local, ctx.GetSessionVars().Location())
        if err != nil {
            return d, err
        }
    }
}
```

Reply:
make sense, done